ethers-core/src/utils/mod.rs

```diff
@@ -421,6 +421,20 @@ pub fn to_checksum(addr: &Address, chain_id: Option<u8>) -> String {
     })
 }
 
+/// checks if the given address is a valid checksum address
+///
+/// Returns `true` if addr is a valid checksum address, `false` otherwise.
+pub fn verify_checksum(addr: &str, chain_id: Option<u8>) -> bool {
```

Comment:
like this?

```rs
/// Returns a `Result<Address>`: `Ok(Address)` if addr is checksummed, `Err()` otherwise
pub fn parse_checksummed(addr: &str, chain_id: Option<u8>) -> Result<Address>
```

Reply:
function signature is awesome 👌 

I'd write docs similar to this:
```/// Parses an [EIP-1191](https://eips.ethereum.org/EIPS/eip-1191) checksum address. Returns `Ok(address)` if the checksummed address is valid, `Err()` otherwise. If `chain_id` is `None`, falls back to [EIP-55](https://eips.ethereum.org/EIPS/eip-55) address checksum method```

Reply:
@prestwich refactored, pls check it